plugins/@grouparoo/klaviyo/src/lib/export/exportRecord.ts

```diff
@@ -0,0 +1,159 @@
+import { getKnownFieldMap } from "./destinationMappingOptions";
+import { Errors, ExportRecordPluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+
+async function handleGroupsChanges(client, email, oldGroups, newGroups) {
+  const listMapByName = await client.getListMapByName();
+
+  // Unsubscribe from old groups
+  for (const groupName of oldGroups) {
+    // Only if the group exists
+    if (listMapByName.hasOwnProperty(groupName)) {
+      await client.lists.removeMembersFromList({
+        listId: listMapByName[groupName],
+        emails: [email],
+      });
+
+      const result = await client.lists.getAllMembers({
+        groupId: listMapByName[groupName],
+      });
+      if (result.records.length === 0) {
+        await client.lists.deleteList(listMapByName[groupName]);
+      }
+    }
+  }
+
+  // Subscribe to new groups
+  for (const groupName of newGroups) {
+    // Create list if not exists
+    if (!listMapByName.hasOwnProperty(groupName)) {
```

Comment:
This would be moved to `listMethods`

Reply:
You mean the whole function? or these lines?

Reply:
creating the list if it didn't exist is in th `listMethods` in other plugins.